packages/resource/src/index.ts

```diff
@@ -0,0 +1,237 @@
+  options?: CacheOptions<T, S>,
+): [
+  fetcher: ResourceFetcher<S, T, R>,
+  invalidate: (source: S) => void,
+  expired: Accessor<CacheEntry<T, S> | undefined>,
+] {
+  const [expired, setExpired] = createSignal<CacheEntry<T, S>>();
+  const localCache = options?.cache || (cache as Record<string, CacheEntry<T, S>>);
+  const key = options?.storageKey || "solid-cache";
+  const save = () => {
+    try {
+      options?.storage?.setItem(key, JSON.stringify(localCache));
+    } catch (e) {}
+  };
+  const expireTimeout =
+    typeof options?.expires === "function"
+      ? options.expires
+      : () => (options?.expires as number | undefined) || 300000;
+  if (options?.storage) {
+    try {
+      Object.assign(localCache, JSON.parse(options.storage.getItem(key) || "{}"));
+    } catch (e) {}
+  }
+  return [
+    async (s: S, info: ResourceFetcherInfo<T, R>) => {
+      const serialized = serializer(s);
+      const cachedEntry: CacheEntry<T, S> | undefined = Object.hasOwn(localCache, serialized)
+        ? localCache[serialized]
+        : undefined;
+      const now = +new Date();
+      if (cachedEntry && (cachedEntry.ts || now) >= now) {
+        return cachedEntry.data;
+      }
+      const response: T = await fetcher(s, info);
+      const entry = { source: s, data: response } as CacheEntry<T, S>;
+      const expiryDelay = expireTimeout(entry);
+      entry.ts = now + expiryDelay;
+      localCache[serialized] = entry;
+      save();
+      setTimeout(() => {
+        if (Object.hasOwn(localCache, serialized)) {
+          delete localCache[serialized];
+          save();
+          setExpired(entry);
+        }
+      }, expiryDelay);
+      return response;
+    },
+    (s: S) => {
+      delete localCache[serializer(s)];
+      save();
+    },
+    expired,
+  ];
+}
+
+export type RetryOptions = {
+  delay?: number;
+  retries?: number;
+};
+
+/**
+ * **Creates a fetcher that retries multiple times in case of errors**
+ * ```ts
+ * const fetcher = makeRetrying((url) => fetch(url), { retries: 5 });
+ * ```
+ * Receives the fetcher and an optional options object and returns a wrapped fetcher that retries on error after a delay multiple times.
+ *
+ * The optional options object contains the following optional properties:
+ * - `delay` - number of Milliseconds to wait before retrying; default is 5s
+ * - `retries` - number of times a request should be repeated before giving up throwing the last error; default is 3 times
+ */
+export function makeRetrying<S, T, R = unknown>(
+  fetcher: ResourceFetcher<S, T, R>,
+  options: RetryOptions = {},
+): ResourceFetcher<S, T, R> {
+  const delay = options.delay ?? 5000;
+  let retries = options.retries || 3;
+  const retrying: ResourceFetcher<S, T, R> = async (k: S, info: ResourceFetcherInfo<T, R>) => {
+    try {
+      return await fetcher(k, info);
+    } catch (error: unknown) {
+      if (retries-- > 0) {
+        delay && await new Promise(resolve => setTimeout(() => resolve, delay));
```

Comment:
```suggestion
        delay && await new Promise(resolve => setTimeout(resolve, delay));
```
Here, `resolve` will never be called, right?